src/CommunityToolkit.Aspire.Hosting.Sqlite/SqliteResource.cs

```diff
@@ -15,5 +17,26 @@ public class SqliteResource(string name, string databasePath, string databaseFil
     internal string DatabaseFilePath => Path.Combine(DatabasePath, DatabaseFileName);
 
     /// <inheritdoc/>
-    public ReferenceExpression ConnectionStringExpression => ReferenceExpression.Create($"Data Source={DatabaseFilePath};Cache=Shared;Mode=ReadWriteCreate;");
+    public ReferenceExpression ConnectionStringExpression => ReferenceExpression.Create($"Data Source={DatabaseFilePath};Cache=Shared;Mode=ReadWriteCreate;Extensions={JsonSerializer.Serialize(Extensions)}");
+
+    private readonly List<ExtensionMetadata> extensions = [];
+
+    /// <summary>
+    /// Gets the extensions to be loaded into the database.
+    /// </summary>
+    /// <remarks>
+    /// Extensions are not loaded by the hosting integration, the information is provided for the client to load the extensions.
+    /// </remarks>
+    public IReadOnlyCollection<ExtensionMetadata> Extensions => extensions;
+
+    internal void AddExtension(ExtensionMetadata extension) => extensions.Add(extension);
 }
+
+/// <summary>
+/// Represents metadata for an extension to be loaded into a database.
+/// </summary>
+/// <param name="Extension">The name of the extension binary, eg: vec0</param>
+/// <param name="PackageName">The name of the NuGet package. Only required if <paramref name="IsNuGetPackage"/> is <see langword="true" />.</param>
+/// <param name="IsNuGetPackage">Indicates if the extension will be loaded from a NuGet package.</param>
+/// <param name="ExtensionFolder">The folder for the extension. Only required if <paramref name="IsNuGetPackage"/> is <see langword="false" />.</param>
+public record ExtensionMetadata(string Extension, string? PackageName, bool IsNuGetPackage, string? ExtensionFolder);
```

Comment:
```suggestion
public record SqliteExtensionMetadata(string Extension, string? PackageName, bool IsNuGetPackage, string? ExtensionFolder);
```

I prefer `SqliteExtensionMetadata` to `ExtensionMetadata` because it prevents feature conflicts (e.g., PostgreSQL extensions like pgvector).